File: 1523 - Estacionamento Linear/Main.java
import java.util.Scanner;
import java.util.Stack;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int k = sc.nextInt();

        while (n != 0 && k != 0) {
            Stack<Integer> stack = new Stack<>();
            boolean livre = true;

            for (int i = 0; i < n; i++) {
                int cheg = sc.nextInt();
                int saida = sc.nextInt();

                while (!stack.isEmpty() && cheg >= stack.peek()) stack.pop();

                if (!stack.isEmpty() && saida > stack.peek()) livre = false;
                if (stack.size() >= k) livre = false;
                if (livre) stack.push(saida);
            }
            
            System.out.println(livre ? "Sim" : "Nao");
            
            n = sc.nextInt();
            k = sc.nextInt();
        }

        sc.close();
    }
}

File: 1523 - Estacionamento Linear/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

typedef struct Celula {
    int saida;
    struct Celula *prox;
} Celula;

Celula *novaCelula(int x) {
    Celula *nova = (Celula *) malloc (sizeof(Celula));
    nova->saida = x;
    nova->prox = NULL;
    
    return nova;
}

Celula *topo = NULL;
int tamanho = 0;

void inserir(int x) {
    Celula *tmp = novaCelula(x);
    tmp->prox = topo;
    topo = tmp;
    tamanho++;
}

int remover() {
    if (topo == NULL) printf("pilha vazia");

    int resp = topo->saida;
    topo = topo->prox;
    tamanho--;

    return resp;
}

int main() {
    int n, k;
    scanf("%d %d", &n, &k);

    while(n != 0 && k != 0) {
        bool livre = true;

        for (int i = 0; i < n; i++) {
            int cheg, saida;
            scanf("%d %d", &cheg, &saida);

            while (topo != NULL && cheg >= topo->saida) remover();

            if (topo != NULL && saida > topo->saida) livre = false;
            if (tamanho >= k) livre = false;
            if (livre) inserir(saida);
        }

        if (livre) {
            printf("Sim\n");
        } else {
            printf("Nao\n");
        }

        while (topo != NULL) {
            Celula *tmp = topo;
            topo = topo->prox;
            free(tmp);
        }

        scanf("%d %d", &n, &k);
    }

    return 0;
}